pkg/operator/ceph/csi/csidriver.go

```diff
@@ -70,7 +79,8 @@ func (d v1CsiDriver) createCSIDriverInfo(ctx context.Context, clientset kubernet
 
 	// As FSGroupPolicy and AttachRequired fields are immutable, should be set only during create time.
 	// if the request is to change the FSGroupPolicy or AttachRequired, we are deleting the CSIDriver object and creating it.
-	if (driver.Spec.FSGroupPolicy != nil && csiDriver.Spec.FSGroupPolicy != nil && *driver.Spec.FSGroupPolicy != *csiDriver.Spec.FSGroupPolicy) || *driver.Spec.AttachRequired != *csiDriver.Spec.AttachRequired {
+	if (driver.Spec.FSGroupPolicy != nil && csiDriver.Spec.FSGroupPolicy != nil && *driver.Spec.FSGroupPolicy != *csiDriver.Spec.FSGroupPolicy) ||
+		!reflect.DeepEqual(driver.Spec, csiDriver.Spec) {
```

Comment:
not all fields are immutable and few fields are allowed to be changed in Update call, with this change we will always delete and recreate the csidriver object which is not good. we should only delete and recreate only if immutable fields are changed.

Reply:
> not all fields are immutable and few fields are allowed to be changed in Update call, with this change we will always delete and recreate the csidriver object which is not good. we should only delete and recreate only if immutable fields are changed.

I don't think anyone will be changing csidriver object manually for this to matter. 
The change will happen only once.

Reply:
opened tracker https://github.com/rook/rook/issues/12442

Reply:
> > not all fields are immutable and few fields are allowed to be changed in Update call, with this change we will always delete and recreate the csidriver object which is not good. we should only delete and recreate only if immutable fields are changed.
> 
> I don't think anyone will be changing csidriver object manually for this to matter. The change will happen only once.

modified to keep inline with the rest of the check.